src/lib/mina_base/party.ml

```diff
@@ -165,6 +185,54 @@ module Update = struct
     end
   end]
 
+  let gen : t Quickcheck.Generator.t =
+    let open Quickcheck.Let_syntax in
+    let%bind app_state =
+      let%bind fields =
+        let field_gen = Snark_params.Tick.Field.gen in
+        Quickcheck.Generator.list_with_length 8 (Set_or_keep.gen field_gen)
+      in
+      (* won't raise because length is correct *)
+      Quickcheck.Generator.return (Snapp_state.V.of_list_exn fields)
+    in
+    let%bind delegate = Set_or_keep.gen Public_key.Compressed.gen in
+    let%bind verification_key =
+      Set_or_keep.gen
+        (Quickcheck.Generator.return
+           (let data = Pickles.Side_loaded.Verification_key.dummy in
+            let hash = Snapp_account.digest_vk data in
+            { With_hash.data; hash }))
+    in
+    let%bind permissions = Set_or_keep.gen Permissions.gen in
+    let%bind snapp_uri =
+      let uri_gen =
+        Quickcheck.Generator.of_list
+          [ "https://www.example.com"
+          ; "https://www.minaprotocol.com"
+          ; "https://www.gurgle.com"
+          ; "https://faceplant.com"
+          ]
+      in
+      Set_or_keep.gen uri_gen
+    in
+    let%bind token_symbol =
+      let token_gen =
+        Quickcheck.Generator.of_list
+          [ "MINA"; "TOKEN1"; "TOKEN2"; "TOKEN3"; "TOKEN4"; "TOKEN5" ]
+      in
+      Set_or_keep.gen token_gen
+    in
+    let%map timing = Set_or_keep.gen Timing_info.gen in
```

Comment:
Note that this can only be set on newly-created accounts; perhaps it makes sense to pass `succeed` (or e.g. `new_account`, derived from the ledger) here?

Reply:
Done.